File: tests/cmp_suite/prime_bench.cpp
#include <iostream>
#include <cmath>

int main() {
    int limit = 100000;
    int count = 0;
    
    for (int num = 2; num < limit; ++num) {
        bool is_prime = true;
        int max_i = static_cast<int>(std::sqrt(num));
        for (int i = 2; i <= max_i; ++i) {
            if (num % i == 0) {
                is_prime = false;
                break;
            }
        }
        if (is_prime) {
            count++;
        }
    }
    
    std::cout << count << std::endl;
    return 0;
}

File: tests/cmp_suite/prime_bench.rs
fn main() {
    let limit: i32 = 100000;
    let mut count: i32 = 0;
    
    for num in 2..limit {
        let mut is_prime = true;
        let max_i = (num as f64).sqrt() as i32;
        for i in 2..=max_i {
            if num % i == 0 {
                is_prime = false;
                break;
            }
        }
        if is_prime {
            count += 1;
        }
    }
    
    println!("{}", count);
}
